### PR TITLE
:ghost: builtin auth test and binding client tech-debt.

### PR DESCRIPTION
**Pull request by jortel**
auth tech debt improvements.
binding.client tech debt improvements.

### CODE REVIEW

File: auth/builtin.go [hunk 1]
```diff
@@ -62,25 +62,14 @@ func (r *NoAuth) Refresh(refresh string) (token Token, err error) {
...
-func parseToken(requestToken string) (token string, err error) {
-	splitToken := strings.Fields(requestToken)
-	if len(splitToken) != 2 || strings.ToLower(splitToken[0]) != "bearer" {
-		err = liberr.Wrap(&NotValid{Token: requestToken})
-		return
-	}
-	token = splitToken[1]
-	return
-}
-
```
Comment (jortel) [inline auth/builtin.go:75]: **hint**: No reason for parseToken() to be a package function. Consistent with the project, this should be a method on Builtin.
Better for parameter to be a request.

File: auth/builtin.go [hunk 2]
```diff
@@ -199,3 +188,14 @@ func (r *Builtin) NewToken(user string, scopes []string, claims jwt.MapClaims) (
 	signed, err = token.SignedString([]byte(Settings.Auth.Token.Key))
 	return
 }
+
+// parseToken returns the token
```
Comment (jortel) [inline auth/builtin.go:192]: **hint**: No reason for parseToken() to be a package function. Consistent with the project, this should be a method on Builtin.
Simpler for parameter to be a request.

File: auth/builtin_test.go [hunk 3]
```diff
@@ -1,66 +1,62 @@
...
-			}
-			if got != tt.want {
-				t.Errorf("parseToken() = %v, want %v", got, tt.want)
-			}
-		})
+	builtin := Builtin{}
+	for _, test := range tests {
+		token, err := builtin.parseToken(test.request)
+		g.Expect(test.wantErr).To(gomega.Equal(err != nil))
+		g.Expect(test.want).To(gomega.Equal(token))
```
Comment (jortel) [inline auth/builtin_test.go:60]: **hint**: simpler and more consistent with other unit tests in the project.

File: binding/client.go [hunk 4]
```diff
@@ -69,34 +69,28 @@ func (s Path) Inject(p Params) (out string) {
...
-	// baseURL for the nub.
-	baseURL string
-	// addon API token
-	token api.Login
 	// transport
 	transport http.RoundTripper
+	// baseURL for the nub.
+	BaseURL string
+	// login API resource.
+	Login api.Login
```
Comment (jortel) [inline binding/client.go:87]: **hint**: renamed to be more accurate.  This is not a token but instead a Login REST resource.
Also, no value making the not-exported.  Simpler.

File: binding/client.go [hunk 5]
```diff
@@ -69,34 +69,28 @@ func (s Path) Inject(p Params) (out string) {
 }
 
 // NewClient Constructs a new client
-func NewClient(url string, token api.Login) (client *Client) {
+func NewClient(baseURL string) (client *Client) {
```
Comment (mansam) [inline binding/client.go:72]: Is anything using NewClient that needs to be adjusted to match?

**Review by mansam — APPROVED** [review]
LGTM

**Pull request merged**